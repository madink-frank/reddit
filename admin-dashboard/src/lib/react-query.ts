import { QueryClient, DefaultOptions } from '@tanstack/react-query';
import type { ApiError } from '../services/api';

// Default query options
const queryConfig: DefaultOptions = {
  queries: {
    // Stale time - how long data is considered fresh
    staleTime: 5 * 60 * 1000, // 5 minutes
    
    // Cache time - how long data stays in cache after component unmounts
    gcTime: 10 * 60 * 1000, // 10 minutes (formerly cacheTime)
    
    // Retry configuration
    retry: (failureCount, error) => {
      const apiError = error as ApiError;
      
      // Don't retry on 4xx errors (client errors)
      if (apiError.status >= 400 && apiError.status < 500) {
        return false;
      }
      
      // Retry up to 3 times for other errors
      return failureCount < 3;
    },
    
    // Retry delay with exponential backoff
    retryDelay: (attemptIndex) => Math.min(1000 * 2 ** attemptIndex, 30000),
    
    // Refetch on window focus for critical data
    refetchOnWindowFocus: false,
    
    // Refetch on reconnect
    refetchOnReconnect: true,
    
    // Refetch on mount if data is stale
    refetchOnMount: true,
  },
  mutations: {
    // Retry mutations once on network errors
    retry: (failureCount, error) => {
      const apiError = error as ApiError;
      
      // Only retry on network errors
      if (apiError.status === 0 || apiError.code === 'NETWORK_ERROR') {
        return failureCount < 1;
      }
      
      return false;
    },
    
    // Retry delay for mutations
    retryDelay: 1000,
  },
};

// Create query client instance
export const queryClient = new QueryClient({
  defaultOptions: queryConfig,
});

// Query key factory for consistent key generation
export const queryKeys = {
  // Auth keys
  auth: {
    user: ['auth', 'user'] as const,
    adminCheck: ['auth', 'admin-check'] as const,
    verify: ['auth', 'verify'] as const,
  },
  
  // Keywords keys
  keywords: {
    all: ['keywords'] as const,
    lists: () => [...queryKeys.keywords.all, 'list'] as const,
    list: (params?: Record<string, unknown>) => [...queryKeys.keywords.lists(), params] as const,
    details: () => [...queryKeys.keywords.all, 'detail'] as const,
    detail: (id: number) => [...queryKeys.keywords.details(), id] as const,
    stats: () => [...queryKeys.keywords.all, 'stats'] as const,
    stat: (id: number) => [...queryKeys.keywords.stats(), id] as const,
  },
  
  // Posts keys
  posts: {
    all: ['posts'] as const,
    lists: () => [...queryKeys.posts.all, 'list'] as const,
    list: (params?: Record<string, unknown>) => [...queryKeys.posts.lists(), params] as const,
    details: () => [...queryKeys.posts.all, 'detail'] as const,
    detail: (id: number) => [...queryKeys.posts.details(), id] as const,
    trending: (params?: Record<string, unknown>) => [...queryKeys.posts.all, 'trending', params] as const,
    search: (query: string, params?: Record<string, unknown>) => [...queryKeys.posts.all, 'search', query, params] as const,
    byKeyword: (keywordId: number, params?: Record<string, unknown>) => [...queryKeys.posts.all, 'keyword', keywordId, params] as const,
    bySubreddit: (subreddit: string, params?: Record<string, unknown>) => [...queryKeys.posts.all, 'subreddit', subreddit, params] as const,
    stats: () => [...queryKeys.posts.all, 'stats'] as const,
    subreddits: () => [...queryKeys.posts.all, 'subreddits'] as const,
  },
  
  // Crawling keys
  crawling: {
    all: ['crawling'] as const,
    status: () => [...queryKeys.crawling.all, 'status'] as const,
    job: (jobId: string) => [...queryKeys.crawling.all, 'job', jobId] as const,
    history: (params?: Record<string, unknown>) => [...queryKeys.crawling.all, 'history', params] as const,
    stats: () => [...queryKeys.crawling.all, 'stats'] as const,
    queue: () => [...queryKeys.crawling.all, 'queue'] as const,
    logs: (jobId: string) => [...queryKeys.crawling.all, 'logs', jobId] as const,
  },
  
  // Analytics keys
  analytics: {
    all: ['analytics'] as const,
    dashboard: () => [...queryKeys.analytics.all, 'dashboard'] as const,
    trends: (params?: Record<string, unknown>) => [...queryKeys.analytics.all, 'trends', params] as const,
    keyword: (id: number) => [...queryKeys.analytics.all, 'keyword', id] as const,
    timeSeries: (params?: Record<string, unknown>) => [...queryKeys.analytics.all, 'timeseries', params] as const,
    subredditDistribution: (params?: Record<string, unknown>) => [...queryKeys.analytics.all, 'subreddit-distribution', params] as const,
    sentiment: (params?: Record<string, unknown>) => [...queryKeys.analytics.all, 'sentiment', params] as const,
    engagement: (params?: Record<string, unknown>) => [...queryKeys.analytics.all, 'engagement', params] as const,
    correlations: (keywordId: number) => [...queryKeys.analytics.all, 'correlations', keywordId] as const,
    heatmap: (params?: Record<string, unknown>) => [...queryKeys.analytics.all, 'heatmap', params] as const,
    compare: (keywordIds: number[]) => [...queryKeys.analytics.all, 'compare', keywordIds] as const,
  },
  
  // Content keys
  content: {
    all: ['content'] as const,
    lists: () => [...queryKeys.content.all, 'list'] as const,
    list: (params?: Record<string, unknown>) => [...queryKeys.content.lists(), params] as const,
    details: () => [...queryKeys.content.all, 'detail'] as const,
    detail: (id: number) => [...queryKeys.content.details(), id] as const,
    generation: (jobId: string) => [...queryKeys.content.all, 'generation', jobId] as const,
    templates: () => [...queryKeys.content.all, 'templates'] as const,
    template: (id: number) => [...queryKeys.content.all, 'template', id] as const,
    stats: () => [...queryKeys.content.all, 'stats'] as const,
    history: (id: number) => [...queryKeys.content.all, 'history', id] as const,
  },
} as const;

// Utility functions for cache management
export const cacheUtils = {
  // Invalidate all queries for a specific domain
  invalidateKeywords: () => queryClient.invalidateQueries({ queryKey: queryKeys.keywords.all }),
  invalidatePosts: () => queryClient.invalidateQueries({ queryKey: queryKeys.posts.all }),
  invalidateCrawling: () => queryClient.invalidateQueries({ queryKey: queryKeys.crawling.all }),
  invalidateAnalytics: () => queryClient.invalidateQueries({ queryKey: queryKeys.analytics.all }),
  invalidateContent: () => queryClient.invalidateQueries({ queryKey: queryKeys.content.all }),
  
  // Remove specific queries from cache
  removeKeyword: (id: number) => queryClient.removeQueries({ queryKey: queryKeys.keywords.detail(id) }),
  removePost: (id: number) => queryClient.removeQueries({ queryKey: queryKeys.posts.detail(id) }),
  removeContent: (id: number) => queryClient.removeQueries({ queryKey: queryKeys.content.detail(id) }),
  
  // Prefetch commonly used data
  prefetchDashboard: () => queryClient.prefetchQuery({
    queryKey: queryKeys.analytics.dashboard(),
    staleTime: 2 * 60 * 1000, // 2 minutes
  }),
  
  // Clear all cache
  clearAll: () => queryClient.clear(),
  
  // Get cached data without triggering a fetch
  getKeywords: () => queryClient.getQueryData(queryKeys.keywords.lists()),
  getPosts: (params?: Record<string, unknown>) => queryClient.getQueryData(queryKeys.posts.list(params)),
  getDashboard: () => queryClient.getQueryData(queryKeys.analytics.dashboard()),
};

// Error handling utilities
export const queryErrorHandler = (error: unknown) => {
  const apiError = error as ApiError;
  
  // Log error for debugging
  console.error('Query Error:', apiError);
  
  // Handle specific error cases
  if (apiError.status === 401) {
    // Handle authentication errors
    window.dispatchEvent(new CustomEvent('auth:logout', { 
      detail: { reason: 'unauthorized' } 
    }));
  } else if (apiError.status === 403) {
    // Handle authorization errors
    console.warn('Access denied:', apiError.message);
  } else if (apiError.status === 0) {
    // Handle network errors
    console.error('Network error - check connection');
  }
  
  return apiError;
};

// Optimistic update helpers
export const optimisticUpdates = {
  // Update keyword in cache optimistically
  updateKeyword: (id: number, updates: Partial<any>) => {
    queryClient.setQueryData(queryKeys.keywords.detail(id), (old: any) => ({
      ...old,
      ...updates,
    }));
  },
  
  // Add new keyword to list optimistically
  addKeyword: (newKeyword: any) => {
    queryClient.setQueryData(queryKeys.keywords.lists(), (old: any) => {
      if (!old) return old;
      return {
        ...old,
        data: [newKeyword, ...old.data],
        total: old.total + 1,
      };
    });
  },
  
  // Remove keyword from list optimistically
  removeKeyword: (id: number) => {
    queryClient.setQueryData(queryKeys.keywords.lists(), (old: any) => {
      if (!old) return old;
      return {
        ...old,
        data: old.data.filter((item: any) => item.id !== id),
        total: old.total - 1,
      };
    });
  },
};