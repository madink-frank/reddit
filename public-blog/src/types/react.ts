// React-specific type definitions

import type { ReactNode, ComponentType } from 'react';

// ============================================================================
// Common React Props
// ============================================================================

export interface BaseProps {
  className?: string;
  children?: ReactNode;
  id?: string;
  'data-testid'?: string;
}

export interface ClickableProps {
  onClick?: (event: React.MouseEvent) => void;
  onKeyDown?: (event: React.KeyboardEvent) => void;
  disabled?: boolean;
  'aria-label'?: string;
}

// ============================================================================
// Component Props
// ============================================================================

export interface ButtonProps extends BaseProps, ClickableProps {
  variant?: 'primary' | 'secondary' | 'outline' | 'ghost' | 'danger';
  size?: 'sm' | 'md' | 'lg';
  loading?: boolean;
  icon?: ReactNode;
  iconPosition?: 'left' | 'right';
  fullWidth?: boolean;
  type?: 'button' | 'submit' | 'reset';
}

export interface InputProps extends BaseProps {
  type?: 'text' | 'email' | 'password' | 'search' | 'url' | 'tel';
  placeholder?: string;
  value?: string;
  defaultValue?: string;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onFocus?: (event: React.FocusEvent<HTMLInputElement>) => void;
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
  disabled?: boolean;
  required?: boolean;
  error?: string;
  label?: string;
  helperText?: string;
  icon?: ReactNode;
  iconPosition?: 'left' | 'right';
}

export interface TextareaProps extends BaseProps {
  placeholder?: string;
  value?: string;
  defaultValue?: string;
  onChange?: (event: React.ChangeEvent<HTMLTextAreaElement>) => void;
  onFocus?: (event: React.FocusEvent<HTMLTextAreaElement>) => void;
  onBlur?: (event: React.FocusEvent<HTMLTextAreaElement>) => void;
  disabled?: boolean;
  required?: boolean;
  error?: string;
  label?: string;
  helperText?: string;
  rows?: number;
  resize?: 'none' | 'vertical' | 'horizontal' | 'both';
}

export interface SelectProps extends BaseProps {
  value?: string;
  defaultValue?: string;
  onChange?: (value: string) => void;
  disabled?: boolean;
  required?: boolean;
  error?: string;
  label?: string;
  helperText?: string;
  placeholder?: string;
  options: SelectOption[];
}

export interface SelectOption {
  value: string;
  label: string;
  disabled?: boolean;
  group?: string;
}

export interface CheckboxProps extends BaseProps {
  checked?: boolean;
  defaultChecked?: boolean;
  onChange?: (checked: boolean) => void;
  disabled?: boolean;
  required?: boolean;
  label?: string;
  error?: string;
  indeterminate?: boolean;
}

export interface RadioProps extends BaseProps {
  checked?: boolean;
  defaultChecked?: boolean;
  onChange?: (checked: boolean) => void;
  disabled?: boolean;
  required?: boolean;
  label?: string;
  error?: string;
  name: string;
  value: string;
}

export interface ModalProps extends BaseProps {
  open: boolean;
  onClose: () => void;
  title?: string;
  size?: 'sm' | 'md' | 'lg' | 'xl' | 'full';
  closeOnOverlayClick?: boolean;
  closeOnEscape?: boolean;
  showCloseButton?: boolean;
  footer?: ReactNode;
}

export interface TooltipProps extends BaseProps {
  content: ReactNode;
  placement?: 'top' | 'bottom' | 'left' | 'right';
  trigger?: 'hover' | 'click' | 'focus';
  delay?: number;
  arrow?: boolean;
}

export interface DropdownProps extends BaseProps {
  trigger: ReactNode;
  items: DropdownItem[];
  placement?: 'bottom-start' | 'bottom-end' | 'top-start' | 'top-end';
  closeOnItemClick?: boolean;
}

export interface DropdownItem {
  key: string;
  label: ReactNode;
  onClick?: () => void;
  disabled?: boolean;
  divider?: boolean;
  icon?: ReactNode;
}

export interface TabsProps extends BaseProps {
  defaultValue?: string;
  value?: string;
  onChange?: (value: string) => void;
  orientation?: 'horizontal' | 'vertical';
  variant?: 'default' | 'pills' | 'underline';
  items: TabItem[];
}

export interface TabItem {
  key: string;
  label: ReactNode;
  content: ReactNode;
  disabled?: boolean;
  icon?: ReactNode;
}

export interface AccordionProps extends BaseProps {
  items: AccordionItem[];
  multiple?: boolean;
  defaultOpenItems?: string[];
  openItems?: string[];
  onOpenChange?: (openItems: string[]) => void;
}

export interface AccordionItem {
  key: string;
  title: ReactNode;
  content: ReactNode;
  disabled?: boolean;
  icon?: ReactNode;
}

export interface PaginationProps extends BaseProps {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
  showFirstLast?: boolean;
  showPrevNext?: boolean;
  maxVisiblePages?: number;
  size?: 'sm' | 'md' | 'lg';
}

export interface BreadcrumbProps extends BaseProps {
  items: BreadcrumbItem[];
  separator?: ReactNode;
  maxItems?: number;
}

export interface BreadcrumbItem {
  label: ReactNode;
  href?: string;
  onClick?: () => void;
  current?: boolean;
}

export interface AlertProps extends BaseProps {
  variant?: 'info' | 'success' | 'warning' | 'error';
  title?: string;
  closable?: boolean;
  onClose?: () => void;
  icon?: ReactNode;
  action?: ReactNode;
}

export interface BadgeProps extends BaseProps {
  variant?: 'default' | 'primary' | 'secondary' | 'success' | 'warning' | 'error';
  size?: 'sm' | 'md' | 'lg';
  dot?: boolean;
  count?: number;
  showZero?: boolean;
}

export interface AvatarProps extends BaseProps {
  src?: string;
  alt?: string;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  shape?: 'circle' | 'square';
  fallback?: ReactNode;
  online?: boolean;
}

export interface CardProps extends BaseProps {
  title?: ReactNode;
  subtitle?: ReactNode;
  image?: string;
  imageAlt?: string;
  actions?: ReactNode;
  hoverable?: boolean;
  loading?: boolean;
}

export interface SkeletonProps extends BaseProps {
  width?: string | number;
  height?: string | number;
  variant?: 'text' | 'rectangular' | 'circular';
  animation?: 'pulse' | 'wave' | false;
  lines?: number;
}

export interface SpinnerProps extends BaseProps {
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  color?: string;
  thickness?: number;
}

// ============================================================================
// Layout Props
// ============================================================================

export interface ContainerProps extends BaseProps {
  maxWidth?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | false;
  fluid?: boolean;
  centerContent?: boolean;
}

export interface GridProps extends BaseProps {
  columns?: number | { xs?: number; sm?: number; md?: number; lg?: number; xl?: number };
  gap?: number | string;
  alignItems?: 'start' | 'center' | 'end' | 'stretch';
  justifyContent?: 'start' | 'center' | 'end' | 'between' | 'around' | 'evenly';
}

export interface FlexProps extends BaseProps {
  direction?: 'row' | 'column' | 'row-reverse' | 'column-reverse';
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
  alignItems?: 'start' | 'center' | 'end' | 'stretch' | 'baseline';
  justifyContent?: 'start' | 'center' | 'end' | 'between' | 'around' | 'evenly';
  gap?: number | string;
}

export interface StackProps extends BaseProps {
  direction?: 'horizontal' | 'vertical';
  spacing?: number | string;
  alignItems?: 'start' | 'center' | 'end' | 'stretch';
  justifyContent?: 'start' | 'center' | 'end' | 'between' | 'around' | 'evenly';
  divider?: ReactNode;
}

// ============================================================================
// Form Props
// ============================================================================

export interface FormProps extends BaseProps {
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
  noValidate?: boolean;
  autoComplete?: 'on' | 'off';
}

export interface FormFieldProps extends BaseProps {
  label?: string;
  error?: string;
  helperText?: string;
  required?: boolean;
  disabled?: boolean;
}

export interface FormGroupProps extends BaseProps {
  title?: string;
  description?: string;
}

// ============================================================================
// Higher-Order Component Types
// ============================================================================

export interface WithLoadingProps {
  loading?: boolean;
  loadingComponent?: ComponentType;
}

export interface WithErrorProps {
  error?: string | Error | null;
  errorComponent?: ComponentType<{ error: string | Error }>;
}

export interface WithAuthProps {
  requireAuth?: boolean;
  fallback?: ComponentType;
}

// ============================================================================
// Hook Return Types
// ============================================================================

export interface UseToggleReturn {
  value: boolean;
  toggle: () => void;
  setTrue: () => void;
  setFalse: () => void;
}

export interface UseLocalStorageReturn<T> {
  value: T;
  setValue: (value: T | ((prev: T) => T)) => void;
  removeValue: () => void;
}

export interface UseDebounceReturn<T> {
  debouncedValue: T;
  cancel: () => void;
  flush: () => void;
}

export interface UsePaginationReturn {
  currentPage: number;
  totalPages: number;
  hasNext: boolean;
  hasPrev: boolean;
  goToPage: (page: number) => void;
  nextPage: () => void;
  prevPage: () => void;
  goToFirst: () => void;
  goToLast: () => void;
}

// ============================================================================
// Event Handler Types
// ============================================================================

export type ClickHandler = (event: React.MouseEvent) => void;
export type ChangeHandler<T = string> = (value: T) => void;
export type SubmitHandler = (event: React.FormEvent) => void;
export type KeyboardHandler = (event: React.KeyboardEvent) => void;
export type FocusHandler = (event: React.FocusEvent) => void;