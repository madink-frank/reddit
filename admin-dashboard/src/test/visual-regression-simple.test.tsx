import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import { BrowserRouter } from 'react-router-dom'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'

// Test wrapper with providers
const TestWrapper = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: { retry: false },
      mutations: { retry: false },
    },
  })

  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <div className="min-h-screen bg-gray-50">
          {children}
        </div>
      </BrowserRouter>
    </QueryClientProvider>
  )
}

describe('Visual Regression Tests - Simple', () => {
  it('should render test wrapper correctly', () => {
    const { container } = render(
      <TestWrapper>
        <div data-testid="test-content">Test Content</div>
      </TestWrapper>
    )
    
    expect(container.querySelector('[data-testid="test-content"]')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render basic HTML elements consistently', () => {
    const { container } = render(
      <TestWrapper>
        <div className="p-4 bg-white rounded-lg shadow">
          <h1 className="text-2xl font-bold text-gray-900">Test Title</h1>
          <p className="text-gray-600 mt-2">Test description text</p>
          <button className="mt-4 px-4 py-2 bg-blue-600 text-white rounded hover:bg-blue-700">
            Test Button
          </button>
        </div>
      </TestWrapper>
    )
    
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render form elements consistently', () => {
    const { container } = render(
      <TestWrapper>
        <form className="space-y-4 p-6 bg-white rounded-lg">
          <div>
            <label htmlFor="test-input" className="block text-sm font-medium text-gray-700">
              Test Input
            </label>
            <input
              id="test-input"
              type="text"
              className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md"
              placeholder="Enter text"
            />
          </div>
          <div>
            <label htmlFor="test-select" className="block text-sm font-medium text-gray-700">
              Test Select
            </label>
            <select
              id="test-select"
              className="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md"
            >
              <option value="">Select option</option>
              <option value="option1">Option 1</option>
              <option value="option2">Option 2</option>
            </select>
          </div>
        </form>
      </TestWrapper>
    )
    
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render grid layouts consistently', () => {
    const { container } = render(
      <TestWrapper>
        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 p-4">
          {Array.from({ length: 6 }).map((_, i) => (
            <div key={i} className="p-4 bg-white rounded-lg shadow">
              <h3 className="font-semibold">Card {i + 1}</h3>
              <p className="text-gray-600 mt-2">Card content</p>
            </div>
          ))}
        </div>
      </TestWrapper>
    )
    
    expect(container.firstChild).toMatchSnapshot()
  })
})