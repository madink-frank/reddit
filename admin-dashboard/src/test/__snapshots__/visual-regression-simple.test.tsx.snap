// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Visual Regression Tests - Simple > should render basic HTML elements consistently 1`] = `
<div
  class="min-h-screen bg-gray-50"
>
  <div
    class="p-4 bg-white rounded-lg shadow"
  >
    <h1
      class="text-2xl font-bold text-gray-900"
    >
      Test Title
    </h1>
    <p
      class="text-gray-600 mt-2"
    >
      Test description text
    </p>
    <button
      class="mt-4 px-4 py-2 bg-blue-600 text-white rounded hover:bg-blue-700"
    >
      Test Button
    </button>
  </div>
</div>
`;

exports[`Visual Regression Tests - Simple > should render form elements consistently 1`] = `
<div
  class="min-h-screen bg-gray-50"
>
  <form
    class="space-y-4 p-6 bg-white rounded-lg"
  >
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="test-input"
      >
        Test Input
      </label>
      <input
        class="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md"
        id="test-input"
        placeholder="Enter text"
        type="text"
      />
    </div>
    <div>
      <label
        class="block text-sm font-medium text-gray-700"
        for="test-select"
      >
        Test Select
      </label>
      <select
        class="mt-1 block w-full px-3 py-2 border border-gray-300 rounded-md"
        id="test-select"
      >
        <option
          value=""
        >
          Select option
        </option>
        <option
          value="option1"
        >
          Option 1
        </option>
        <option
          value="option2"
        >
          Option 2
        </option>
      </select>
    </div>
  </form>
</div>
`;

exports[`Visual Regression Tests - Simple > should render grid layouts consistently 1`] = `
<div
  class="min-h-screen bg-gray-50"
>
  <div
    class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 p-4"
  >
    <div
      class="p-4 bg-white rounded-lg shadow"
    >
      <h3
        class="font-semibold"
      >
        Card 
        1
      </h3>
      <p
        class="text-gray-600 mt-2"
      >
        Card content
      </p>
    </div>
    <div
      class="p-4 bg-white rounded-lg shadow"
    >
      <h3
        class="font-semibold"
      >
        Card 
        2
      </h3>
      <p
        class="text-gray-600 mt-2"
      >
        Card content
      </p>
    </div>
    <div
      class="p-4 bg-white rounded-lg shadow"
    >
      <h3
        class="font-semibold"
      >
        Card 
        3
      </h3>
      <p
        class="text-gray-600 mt-2"
      >
        Card content
      </p>
    </div>
    <div
      class="p-4 bg-white rounded-lg shadow"
    >
      <h3
        class="font-semibold"
      >
        Card 
        4
      </h3>
      <p
        class="text-gray-600 mt-2"
      >
        Card content
      </p>
    </div>
    <div
      class="p-4 bg-white rounded-lg shadow"
    >
      <h3
        class="font-semibold"
      >
        Card 
        5
      </h3>
      <p
        class="text-gray-600 mt-2"
      >
        Card content
      </p>
    </div>
    <div
      class="p-4 bg-white rounded-lg shadow"
    >
      <h3
        class="font-semibold"
      >
        Card 
        6
      </h3>
      <p
        class="text-gray-600 mt-2"
      >
        Card content
      </p>
    </div>
  </div>
</div>
`;

exports[`Visual Regression Tests - Simple > should render test wrapper correctly 1`] = `
<div
  class="min-h-screen bg-gray-50"
>
  <div
    data-testid="test-content"
  >
    Test Content
  </div>
</div>
`;
